File: Backend/Domain/Entities/Book.cs
using System;
using Domain.Common;

namespace Domain.Entities
{
	public class Book: BaseEntity
	{
		public string Title { get; set; }
		public DateTime PublishedDate { get; set; }
		public Guid AuthorId { get; set; }
		public bool IsAvailable { get; set; }

        public Author Author { get; set; }
    }
}

File: library-web/src/app/domain/models/book.ts
import { Author } from "./author";

export interface BookResponse {
    id: string;
    dateAdded: Date;
    title: string;
    publishedDate: Date;
    authorId: string;
    isAvailable: boolean;
    author: Author;
}

export interface BookRequest {
    title: string;
    publishedDate: Date;
    authorId: string;
    isAvailable: boolean;
}